org.eclipse.jdt.debug/model/org/eclipse/jdt/internal/debug/core/model/JDIDebugTarget.java

```diff
@@ -338,6 +338,10 @@ public class JDIDebugTarget extends JDIDebugElement implements
 	 */
 	private final Map<Long, String> objectLabels = new HashMap<>();
 
+	/**
+	 * Preference storage for active debug session.
+	 */
+	public static Map<String,Object> debugSessionVariables = new HashMap<>();
```

Comment:
if this can be accessed from different threads it should be a final ConcurrentHashMap

Reply:
okay